components/UI/ReasoningFilterMenu/index.tsx

```diff
@@ -63,17 +64,39 @@ const ReasoningFilterMenu = ({ onChange }: ReasoningFilterMenuProps) => {
     const value = event.currentTarget.dataset.value as string;
 
     setSelectedValue(value);
-
+    window.localStorage.setItem(
+      localStorageKeys.reasonFilter,
+      JSON.stringify(valueToOption(value))
+    );
     handleClose();
   };
 
+  const getLabel = (
+    options: readonly ReasoningFilterMenuOption[],
+    value: string
+  ) => {
+    const option = options?.find((option) => option.value === value);
+    return option?.label;
+  };
+
   useEffect(() => {
     const option = valueToOption(selectedValue);
 
     onChange(option);
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [selectedValue]);
 
+  useEffect(() => {
+    const reasonFilter = window.localStorage.getItem(
```

Comment:
Why do we keep the user's notification preferences in localStorage? This would be like saving your last search on the search page on an e-commerce site. I think we should just keep it in the url. We don't need recent notification preferences when tab is closed.

I may be wrong, it would be better if you explain your reasoning for using localStorage.

Reply:
i think my pr #740 is better.